### DISCUSSION TITLE
Fallback for route string

### DISCUSSION THREAD

**Discussion by NilsJacobsen** [vikejs/vike#1615] [Discussion] [answered]
Hi @brillout, 

I'm currently working with a route string and I have the following probem.

```ts
// /pages/m/@id/@slug
 
// Matches the URLs:
//  /m/sjdhfksd/redirected-slug/
//  /m/id8eyjdh/redirected-slug/
//  /m/id8eyjdh/redirected-slug/blog/getstarted
//  ...
export default '/m/@id/@slug/*'
```

We use the id to identify the item and always redirect to the correct slug for SEO reasons. I'm making the configuration in the `+onBeforeRender` file. 

When I have this url -> `/pages/m/@id`, it doesn't get picked by the route and leads to the general `_error` page. What I want is for it to redirect to the slug, which is defined by the matched item (with the ID). Can I somehow create that fallback?

So this `/pages/m/sdfhskdh` shouldn't lead to an error because the item can be identified. It should redirect to -> `/pages/m/sdfhskdh/redirected-slug`

Any ideas or workarounds on how to solve it?

**Comment by brillout** [comment, marked as answer]
How about using a [Route Function](https://vike.dev/route-function) with [`throw redirect()`](https://vike.dev/redirect)? Alternatively you can use [`*`](https://vike.dev/route-string#globs) instead of `@slug` to redirect every path after `/@id/` (in case that's something you want).

**Reply by NilsJacobsen** [reply]
I'll probably use the `/@id/` approach at first. Thanks, for the quick answer.

**Reply by brillout** [reply]
Also see the improved globbing docs: https://vike.dev/route-string#globs. And also make sure to use the latest version which contains a small fix around globbing.

**Reply by NilsJacobsen** [reply]
Thanks, Rom 👍
